### PR TITLE
[Heartbeat] Trace socket

### PR DESCRIPTION
**Pull request by andrewvc**
Add preliminary ability to trace execution via a unix socket.

No release note because this is a tool more for internal use / debugging

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: heartbeat/beater/heartbeat.go [hunk 1]
```diff
@@ -74,6 +76,18 @@ func New(b *beat.Beat, rawConfig *conf.C) (beat.Beater, error) {
 		stateLoader, replaceStateLoader = monitorstate.DeferredStateLoader(monitorstate.NilStateLoader, 15*time.Second)
 	}
 
+	var trace tracer.Tracer = tracer.NewNoopTracer()
+	stConfig := parsedConfig.SocketTrace
+	if stConfig != nil {
+		// Note this, intentionally, blocks until connected to the trace endpoint
+		var err error
+		logp.L().Info("Setting up sock tracer")
```
Comment (vigneshshanmugam) [inline heartbeat/beater/heartbeat.go:84]: nit: lets also log the path

File: heartbeat/tracer/tracer.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	for {
+		elapsed := time.Since(started)
+		if elapsed > wait {
+			return st, fmt.Errorf("wait time for sock trace exceeded: %s", wait)
+		}
+		if _, err := os.Stat(st.path); err == nil {
+			logp.L().Info("socktracer found file for unix socket: %s, will attempt to connect")
+			break
+		} else {
+			delay := time.Millisecond * 250
```
Comment (vigneshshanmugam) [inline heartbeat/tracer/tracer.go:52]: nit: can be moved up.

File: x-pack/heartbeat/package-lock.json
Comment (vigneshshanmugam) [inline x-pack/heartbeat/package-lock.json:1]: nit: need to remove this file.

File: x-pack/heartbeat/monitors.d/sample.tcp.yml.disabled
Comment (vigneshshanmugam) [inline x-pack/heartbeat/monitors.d/sample.tcp.yml.disabled:1]: Not sure, how these files are generated under `monitors.d` folder.

**Review by vigneshshanmugam — APPROVED** [review]
LGTM 🎉

File: heartbeat/beater/heartbeat.go (hunk 1 above)
Comment (andrewvc) [reply heartbeat/beater/heartbeat.go]: Done!

**Pull request merged**